Common/Securities/CurrencyConversion/SecurityCurrencyConversion.cs

```diff
@@ -0,0 +1,268 @@
+                {
+                    steps.Add(new Step(existingSecurity, inverted));
+                }
+                else
+                {
+                    steps.Add(new Step(makeNewSecurity(potentialConversionRateSymbol), inverted));
+                }
+
+                return new SecurityCurrencyConversion(sourceCurrency, destinationCurrency, steps);
+            }
+
+            // Search for 2 leg conversions
+            foreach (var potentialConversionRateSymbol1 in allSymbols)
+            {
+                if (!potentialConversionRateSymbol1.PairContainsCurrency(sourceCurrency))
+                {
+                    continue;
+                }
+
+                var middleCurrency = potentialConversionRateSymbol1.CurrencyPairDual(sourceCurrency);
+
+                foreach (var potentialConversionRateSymbol2 in allSymbols)
+                {
+                    if (!potentialConversionRateSymbol2.PairContainsCurrency(middleCurrency))
+                    {
+                        continue;
+                    }
+
+                    if (potentialConversionRateSymbol2.CurrencyPairDual(middleCurrency) != destinationCurrency)
+                    {
+                        continue;
+                    }
+
+                    var steps = new List<Step>(2);
+
+                    string baseCurrency;
+                    string quoteCurrency;
+
+                    CurrencyPairUtil.DecomposeCurrencyPair(
+                        potentialConversionRateSymbol1,
+                        out baseCurrency,
+                        out quoteCurrency);
+
+                    // Step 1
+                    Security existingSecurity1;
+                    if (securitiesBySymbol.TryGetValue(potentialConversionRateSymbol1, out existingSecurity1))
+                    {
+                        steps.Add(new Step(existingSecurity1, sourceCurrency == quoteCurrency));
+                    }
+                    else
+                    {
+                        steps.Add(
+                            new Step(makeNewSecurity(potentialConversionRateSymbol1), sourceCurrency == quoteCurrency));
+                    }
+
+                    CurrencyPairUtil.DecomposeCurrencyPair(
+                        potentialConversionRateSymbol2,
+                        out baseCurrency,
+                        out quoteCurrency);
+
+                    // Step 2
+                    Security existingSecurity2;
+                    if (securitiesBySymbol.TryGetValue(potentialConversionRateSymbol2, out existingSecurity2))
+                    {
+                        steps.Add(new Step(existingSecurity2, middleCurrency == quoteCurrency));
+                    }
+                    else
+                    {
+                        steps.Add(
+                            new Step(makeNewSecurity(potentialConversionRateSymbol2), middleCurrency == quoteCurrency));
+                    }
```

Comment:
We could add some private helper method that performs this `if/else` which we do 3 times